tests/UI/pages/BO/orders/view/tabListBlock.js

```diff
@@ -0,0 +1,461 @@
+
+    return fileName.replace('#', '').trim();
+  }
+
+  /**
+   * Get document name
+   * @param page {Page} Browser tab
+   * @param rowChild {number} Document row on table
+   * @returns {Promise<string>}
+   */
+  async getDocumentType(page, rowChild = 1) {
+    await this.goToDocumentsTab(page);
+
+    return this.getTextContent(page, this.documentType(rowChild));
+  }
+
+  /**
+   * Download a document in document tab
+   * @param page {Page} Browser tab
+   * @param row {number} Document row on table
+   * @return {Promise<string>}
+   */
+  downloadDocument(page, row) {
+    return this.clickAndWaitForDownload(page, this.documentNumberLink(row));
+  }
+
+  /**
+   * Download invoice
+   * @param page {Page} Browser tab
+   * @param row {number} Row of the invoice
+   * @returns {Promise<void>}
+   */
+  async downloadInvoice(page, row = 1) {
+    await this.goToDocumentsTab(page);
+
+    return this.downloadDocument(page, row);
+  }
+
+  /**
+   * Set document note
+   * @param page {Page} Browser tab
+   * @param note {String} Text to set on note input
+   * @param row {number} Row in documents table
+   * @returns {Promise<string>}
+   */
+  async setDocumentNote(page, note, row = 1) {
+    await this.waitForSelectorAndClick(page, this.addDocumentNoteButton(row));
+    await this.setValue(page, this.documentNoteInput(row + 1), note);
+    await this.waitForSelectorAndClick(page, this.documentNoteSaveButton(row + 1));
+
+    return this.getAlertSuccessBlockParagraphContent(page);
+  }
+
+  /**
+   * Is edit note button visible
+   * @param page {Page} Browser tab
+   * @param row {number} Row on table documents
+   * @returns {Promise<boolean>}
+   */
+  async isEditDocumentNoteButtonVisible(page, row = 1) {
+    await this.goToDocumentsTab(page);
+
+    return this.elementVisible(page, this.editDocumentNoteButton(row), 1000);
+  }
+
+  /**
+   * Is add note button visible
+   * @param page {Page} Browser tab
+   * @param row {number} Row on table documents
+   * @returns {Promise<boolean>}
+   */
+  async isAddDocumentNoteButtonVisible(page, row = 1) {
+    await this.goToDocumentsTab(page);
+
+    return this.elementVisible(page, this.addDocumentNoteButton(row), 1000);
+  }
+
+  /**
+   * Is enter payment button visible
+   * @param page {Page} Browser tab
+   * @param row {number} Row on table documents
+   * @returns {Promise<boolean>}
+   */
+  async isEnterPaymentButtonVisible(page, row = 1) {
+    await this.goToDocumentsTab(page);
+
+    return this.elementVisible(page, this.enterPaymentButton(row), 1000);
+  }
+
+  /**
+   * Click on enter payment button and get amount value
+   * @param page {Page} Browser tab
+   * @param row {number} Row on table
+   * @returns {Promise<*>}
+   */
+  async clickOnEnterPaymentButton(page, row = 1) {
+    await this.waitForSelectorAndClick(page, this.enterPaymentButton(row));
+
+    return page.$eval(paymentBlock.paymentAmountInput, el => el.value);
+  }
```

Comment:
Maybe split this in 2 functions, so you don't call another block here

```js
// tabListBlock.js
async clickOnEnterPaymentButton(page, row = 1) {
  await this.waitForSelectorAndClick(page, this.enterPaymentButton(row));
}


// PaymentBlock.js
getPaymentAmountInputValue(page) {
   return page.$eval(this.paymentAmountInput, el => el.value);
}
```

And On the test 
```js
await viewOrderPageTabList.clickOnEnterPaymentButton(page);
const amountValue = await viewOrderPageTabList.getPaymentAmountInputValue(page);
await expect(amountValue).to.not.equal('');
```

Reply:
Done thx :+1: